app/main.py

```diff
@@ -1,8 +1,27 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+        Person.people[name] = self
+
+
+def create_person_list(people: dict) -> list:
+    person_list = []
+
+    for person in people:
+        new_person = Person(person["name"], person["age"])
+        person_list.append(new_person)
+
+    for person in people:
+        new_person = Person.people[person["name"]]
```

Comment:
This loop seems redundant because it iterates over the same `people` dictionary again to access `Person.people`. Consider combining this logic with the first loop to avoid unnecessary iteration.